packages/dd-trace/test/appsec/index.next.plugin.spec.js

```diff
@@ -0,0 +1,232 @@
+const { satisfies } = require('semver')
+const path = require('path')
+
+const { DD_MAJOR } = require('../../../../version')
+const agent = require('../plugins/agent')
+
+describe('test suite', () => {
+  let server
+  let port
+
+  const satisfiesStandalone = version => satisfies(version, '>=12.0.0')
+
+  withVersions('next', 'next', DD_MAJOR >= 4 && '>=11', version => {
+    const realVersion = require(`${__dirname}/../../../../versions/next@${version}`).version()
+
+    function initApp (appName) {
+      const appDir = path.join(__dirname, 'next', appName)
+
+      before(async function () {
+        this.timeout(120 * 1000) // Webpack is very slow and builds on every test run
+
+        const cwd = appDir
+
+        const pkg = require(`${__dirname}/../../../../versions/next@${version}/package.json`)
+
+        if (realVersion.startsWith('10')) {
+          return this.skip() // TODO: Figure out why 10.x tests fail.
+        }
+        delete pkg.workspaces
+
+        // builds fail for next.js 9.5 using node 14 due to webpack issues
+        // note that webpack version cannot be set in v9.5 in next.config.js so we do it here instead
+        // the link below highlights the initial support for webpack 5 (used to fix this issue) in next.js 9.5
+        // https://nextjs.org/blog/next-9-5#webpack-5-support-beta
+        if (realVersion.startsWith('9')) pkg.resolutions = { webpack: '^5.0.0' }
+
+        writeFileSync(`${appDir}/package.json`, JSON.stringify(pkg, null, 2))
+
+        // installing here for standalone purposes, copying `nodules` above was not generating the server file properly
+        // if there is a way to re-use nodules from somewhere in the versions folder, this `execSync` will be reverted
+        execSync('yarn install', { cwd })
+
+        // building in-process makes tests fail for an unknown reason
+        execSync('yarn exec next build', {
+          cwd,
+          env: {
+            ...process.env,
+            version
+          },
+          stdio: ['pipe', 'ignore', 'pipe']
+        })
+
+        if (satisfiesStandalone(realVersion)) {
+          // copy public and static files to the `standalone` folder
+          // const publicOrigin = `${appDir}/public`
+          const publicDestination = `${appDir}/.next/standalone/public`
+          const rulesFileOrigin = `${appDir}/appsec-rules.json`
+          const rulesFileDestination = `${appDir}/.next/standalone/appsec-rules.json`
+
+          execSync(`mkdir ${publicDestination}`)
+          // execSync(`cp ${publicOrigin}/test.txt ${publicDestination}/test.txt`)
+          execSync(`cp ${rulesFileOrigin} ${rulesFileDestination}`)
+        }
+      })
+
+      after(function () {
+        this.timeout(5000)
+        const files = [
+          'package.json',
+          'node_modules',
+          '.next',
+          'yarn.lock'
+        ]
+        const paths = files.map(file => `${appDir}/${file}`)
+        execSync(`rm -rf ${paths.join(' ')}`)
+      })
```

Comment:
These `execSync` could be replaced by functions from `fs` and avoid to create a new shell every time.